### ISSUE TITLE
python implement-matrix partial derive by robot matrix.

### ISSUE THREAD

**Issue by jikan1221** [jikan1221/JK_BLOG#18]
Loss and Invoke fuction
````
def sigmoid(z):
    return 1/(1+np.exp(-(z)))

  def cross_entropy(y,y_hat):
    return -y * np.log(y_hat) - (1 - y) * np.log(1 - y_hat)  
````
````
def cal_loss(X,w,b,loss_func=lambda y_hat:cross_entropy(y,y_hat)):
    z_prod = np.dot(X,w)
    z = z_prod + b * np.ones(z_prod.shape)
    pred_p1 = np.array(list(map(sigmoid,z))).T
    loss = np.mean(np.array(list(map(loss_func,pred_p1))).T,axis=0)
    return loss
````
caculate gradient-->
````
def diff(func,x,step=0.0001):
    diff_mat = step * np.eye(len(x))
    forw_step = x + diff_mat
    forw_val = func(forw_step)
    back_step = x - diff_mat
    back_val = func(back_step)
    differential =  (forw_val - back_val)/(2*step)
    return differential.reshape(len(differential),1) 
````
call 
````
 w_gradient = diff(lambda w: cal_loss(X,w,b),w)
 b_gradient = diff(lambda b: cal_loss(X,w,b),b)
 alpha_temp = alpha/(1 + 0.0000001*i)

    w_new = w - alpha_temp * w_gradient
    b_new = b - alpha_temp * b_gradient


    if (sum(abs(w-w_new)) + sum(abs(b-b_new)))/(N+1) < 10**(-10) or (cur_loss < 0.0001):
        w = w_new
        b = b_new
        break
    else:
        w = w_new
        b = b_new
        i += 1 
````
